src/fparser/two/tests/test_utils.py

```diff
@@ -141,3 +144,30 @@ def test_endstmtbase_match():
         require_stmt_type=True,
     )
     assert result == ("SUBROUTINE", Fortran2003.Name("sub"))
+
+
+@pytest.mark.usefixtures("f2003_create")
+def test_base_include_handling(tmpdir):
```

Comment:
Oh, actually I remember now: it's not as simple as just extending existing tests as they don't make use of the FIFO functionality. I'll see if I can improve this...

Reply:
Weirdly, one of the existing tests did claim to test this functionality but actually didn't. I've updated it so that it reproduces the error on master and thus demonstrates that it's fixed now. In doing this I found that I'd previously misunderstood the bug (not sure how). It seems we don't reverse things but that we lose the first line of an INCLUDEd file if it is pushed back to the FIFO.